### PR TITLE
Update-cellranger-modules

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by fmalmeida — APPROVED** [review]
LGTM once tests pass.

**Comment by grst** [comment]
I'll fix the tests in #4901, then they should hopefully pass here as well.

**Comment by edmundmiller** [comment]
Could we push the images to quay.io?

**Comment by apeltzer** [comment]
I dont mind pushing them there, can do it. Not sure I have the permissions for that though

**Comment by grst** [comment]
ok, that didn't work. #4901 only fixed tests for `cellranger/count`, not the other modules. And even here it now complains that it can't find the docker image.

**Comment by apeltzer** [comment]
@nf-core/core can someone push the images to quay.io please?

**Comment by apeltzer** [comment]
Tests run out after 2 hours 🥹

File: modules/nf-core/cellranger/mkref/tests/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+process {
+    withName: CELLRANGER_MKREF {
+        publishDir = [ path: { "output" } ]
```
Comment (maxulysse) [inline modules/nf-core/cellranger/mkref/tests/nextflow.config:3]: do we need to publish?

File: modules/nf-core/cellranger/mkvdjref/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// nf-core modules test cellranger/mkvdjref
+nextflow_process {
+
+    name "Test Process CELLRANGER_MKVDJREF"
+    script "../main.nf"
+    config "./nextflow.config"
```
Comment (maxulysse) [inline modules/nf-core/cellranger/mkvdjref/tests/main.nf.test:6]:
```suggestion
```

I think it's useless

File: modules/nf-core/cellranger/mkvdjref/tests/nextflow.config
Comment (maxulysse) [inline modules/nf-core/cellranger/mkvdjref/tests/nextflow.config:1]: this file can be deleted

File: modules/nf-core/cellranger/mkref/tests/nextflow.config (hunk 1 above)
Comment (apeltzer) [reply modules/nf-core/cellranger/mkref/tests/nextflow.config:3]: Probably not 😉

Comment (maxulysse) [reply modules/nf-core/cellranger/mkref/tests/nextflow.config:3]: then let's delete the file

Comment (maxulysse) [reply modules/nf-core/cellranger/mkref/tests/nextflow.config:3]: actually we're not publishing in tests, so yeah, let's delete

File: modules/nf-core/cellranger/mkref/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// nf-core modules test cellranger/mkref
+nextflow_process {
+
+    name "Test Process CELLRANGER_MKREF"
+    script "../main.nf"
+    config "./nextflow.config"
```
Comment (maxulysse) [inline modules/nf-core/cellranger/mkref/tests/main.nf.test:6]:
```suggestion
```

File: modules/nf-core/cellranger/mkref/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                input[1] = []
+                input[2] = "homo_sapiens_reference"
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.reference).match() }
```
Comment (maxulysse) [inline modules/nf-core/cellranger/mkref/tests/main.nf.test:64]: what about the versions?

File: modules/nf-core/cellranger/mkvdjref/tests/nextflow.config
Comment (maxulysse) [inline modules/nf-core/cellranger/mkvdjref/tests/nextflow.config:1]: we can delete this one too

File: modules/nf-core/cellranger/multi/tests/main.nf.test
Comment (maxulysse) [inline modules/nf-core/cellranger/multi/tests/main.nf.test:1]: we need to snaphot versions here too

File: modules/nf-core/cellranger/vdj/tests/nextflow.config
Comment (maxulysse) [inline modules/nf-core/cellranger/vdj/tests/nextflow.config:1]: let's delete this one too

**Comment by apeltzer** [comment]
[ x ] Removed all unnecessary `nextflow.config` files
[ x ] Updated tests to include versions for all submodules

(will push one after another, building snapshots on gitpod takes quite long...)

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -207,20 +207,6 @@ jobs:
             tags: bwa/samse
           - profile: conda
             tags: cellpose
-          - profile: conda
-            tags: cellranger/count
```
Comment (fmalmeida) [inline .github/workflows/test.yml:211]: Are we missing `cellranger/count` in the new `test.yml` workflow?

File: modules/nf-core/cellranger/mkref/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                        path( "${outputDir}/homo_sapiens_reference/fasta"                 ),
+                        path( "${outputDir}/homo_sapiens_reference/genes"                 ),
+                        path( "${outputDir}/homo_sapiens_reference/reference.json"        ),
+                        path( "${outputDir}/homo_sapiens_reference/star/SA"               ),
+                        path( "${outputDir}/homo_sapiens_reference/star/SAindex"          ),
+                        path( "${outputDir}/homo_sapiens_reference/star/Genome"           ),
+                        path( "${outputDir}/homo_sapiens_reference/star/geneInfo.tab"     ),
+                        path( "${outputDir}/homo_sapiens_reference/star/exonGeTrInfo.tab" ),
+                        path( "${outputDir}/homo_sapiens_reference/star/exonInfo.tab"     ),
+                        path( "${outputDir}/homo_sapiens_reference/star/sjdbList.out.tab" )
```
Comment (fmalmeida) [inline modules/nf-core/cellranger/mkref/tests/main.nf.test:38]: Are these working now that the publishing was removed?

**Review by fmalmeida — COMMENTED** [review]
Added two questions, because I do not know if they are working or not.
Overall it looks good to me. 
If Maxime's comments were all resolved, I have nothing else to add.

I will wait for the checks to pass before adding approval because it relates to one of my questions.

😄

File: modules/nf-core/cellranger/mkref/tests/main.nf.test (hunk 6 above)
Comment (apeltzer) [reply modules/nf-core/cellranger/mkref/tests/main.nf.test:38]: Haha, no. As I just found out.

File: .github/workflows/test.yml (hunk 5 above)
Comment (apeltzer) [reply .github/workflows/test.yml:211]: No, it was already added there by Adam Talbot ~8 months ago (y)

**Comment by apeltzer** [comment]
Ok @maxulysse  / @fmalmeida - all done and works locally in Gitpod.

**Review by maxulysse — APPROVED** [review]

**Comment by apeltzer** [comment]
See resuls from local / gitpod: 
image

**Comment by grst** [comment]
Here we go! 

This (https://github.com/nf-core/modules/pull/5016/commits/9708c26fa1fc52550a7d83965d3c358442ee3df4) did the trick. Apparently `mkref` has been changed from a simple script to a Martian pipeline between 7.x and 8.x -- therefore we now need to pass the resource arguments.

**Review by grst — APPROVED** [review]

**Comment by fmalmeida** [comment]
Awesome @grst

**Pull request merged**